compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -1235,6 +1235,11 @@ fn collect_items_of_instance<'tcx>(
     };
 
     if mode == CollectionMode::UsedItems {
+        if tcx.sess.opts.debuginfo == DebugInfo::Full {
+            for var_debug_info in &body.var_debug_info {
+                collector.visit_var_debug_info(var_debug_info);
+            }
+        }
```

Comment:
> The collection is limited to full debuginfo builds to match behavior of FunctionCx::compute_per_local_var_debug_info.

That would be good to have as a comment in the code, otherwise someone reading the code will have a really hard time figuring out why this `if` makes sense here.